pynucastro/nucdata/spin_nuclide.py

```diff
@@ -0,0 +1,110 @@
+import os
+import numpy as np
+from pynucastro.nucdata import PeriodicTable
+
+class SpinNuclide:
+    """
+    This class contains the spin information for each nuclei specified in the nubase20.txt table.
+    The purpose to create this class lies on the 
+    """
+
+    def __init__(self, a=None, z=None, spin_states=None):
+
+        self.a = int(a)
+        self.z = int(z)
+        self.spin_states = spin_states
+
+    def __repr__(self):
+        
+        if self.a == 1 and self.z == 0:
+            rep = 'n'
+        else:
+            element = PeriodicTable.lookup_Z(self.z)
+            rep = '{}{}'.format(element.abbreviation, self.a)
+
+        return rep
+
+class SpinTable:
+    """
+    This class stores the nubase2020.txt table information in a dictionary data structure
+    that maps a pynucastro.rates.Nucleus object to a pynucastro.nucdata.SpinNuclide object.
+    Therefore, after setting an SpinTable class in rates.py, we should retrieve the SpinNuclide
+    data structure from a designated Nucleus class.
+    """
+
+    def __init__(self, datafile=None, set_double_gs=False):
+
+        self.set_double_gs = set_double_gs
+        self._spin_nuclide = {}
+
+        if datafile:
+            self.datafile = datafile
+        else:
+            datafile_name = 'nubase2020.txt'
+            nucdata_dir = os.path.dirname(os.path.realpath(__file__))
+            datafile_dir = os.path.join(os.path.join(nucdata_dir, 'AtomicMassEvaluation'), datafile_name)
+
+        if os.path.isfile(datafile_dir):
+            self.datafile = datafile_dir
+        else:
+            raise
```

Comment:
A bare `raise` isn't valid here, you need to pass some sort of exception. You should also only do this inside the else block above.